pallets/runtime/tests/src/settlement_test.rs

```diff
@@ -3274,3 +3244,70 @@ fn dirty_storage_with_tx() {
             );
         });
 }
+
+#[test]
+fn reject_failed_instruction() {
+    ExtBuilder::default().build().execute_with(|| {
+        let alice = User::new(AccountKeyring::Alice);
+        let bob = User::new(AccountKeyring::Bob);
+
+        let token_name = b"ACME";
+        let ticker = Ticker::try_from(&token_name[..]).unwrap();
+        let venue_counter = init(token_name, ticker, AccountKeyring::Alice.public());
+        let amount = 100u128;
+
+        let instruction_counter = create_instruction(&alice, &bob, venue_counter, ticker, amount);
+
+        assert_ok!(Settlement::affirm_instruction(
+            bob.origin(),
+            instruction_counter,
+            default_portfolio_vec(bob.did),
+            1
+        ));
+
+        next_block();
+        assert_eq!(
+            Settlement::instruction_details(instruction_counter).status,
+            InstructionStatus::Failed,
+        );
+
+        next_block();
+        assert_ok!(Settlement::reject_instruction(
+            bob.origin(),
+            instruction_counter,
+            default_portfolio_vec(bob.did),
+            1
+        ));
+
+        next_block();
+        assert_eq!(
+            Settlement::instruction_details(instruction_counter).status,
```

Comment:
Let's extract this into a closure, `let status = || Settlement::instruction_details(instruction_counter).status;` and use that 2x.